### DISCUSSION TITLE
Can Table Action access & edit toogleages and filters columns at initial state ?

### DISCUSSION THREAD

**Discussion by Thiktak** [filamentphp/filament#3404] [Help]
Hello,

Do we have any way for an Action to execute code when the button/link mounted ?

I tried to create an action's Table, but I cannot execute some specific code when the action is mounted.
With `mountUsing` I can customize when I click on the button.

The purpose is to acces to the initialized **Table Component** and read/modify the filters/sorting.
I would like to acces but also changes the values (ex: hidden columns, before the table is rendered (default) or after (data_set)).

It it possible to create an Action with the sames behavior as `x-tables::toggleable` and `x-tables::filters.popover` ?

I tried, mountUsing, configureUsing, mount. Do we have a magical solution ?
My other thought is to use the session (but will work only for toggle).

Thanks.

```php
class ActionVariantSave extends \Filament\Tables\Actions\Action
{
    use CanCustomizeProcess;

    public static function getDefaultName(): ?string {
        return 'Save';
    }

    // make does not work
    // public static function make(?string $name = null): static { return dd(parent::make($name)); }

    protected function setUp(): void {
    	parent::setUp();

        $this->icon('heroicon-o-save'); // ok, I can customize the Action

        // Trigger when click on the button :(
        $this->mountUsing(function (ComponentContainer $form, Model $record): void {
        	dd($this);
        });
    }
}
```

**Comment by danharrin** [comment]
Hey, I'm quite confused by your question, please rephrase it about what you're actually trying to do with this action instead of how you're trying to do it.

https://xyproblem.info
